docs/tutorials/orc.ipynb

```diff
@@ -0,0 +1,336 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "zaiXjZiXzrHs"
+      },
+      "outputs": [],
+      "source": [
+        "!curl -OL https://github.com/tensorflow/io/raw/master/tests/test_orc/iris.orc\n",
+        "!ls -l iris.orc"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "7DG9JTJ0-bzg"
+      },
+      "source": [
+        "## Create a dataset from the file"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ppFAjXAYsj-z"
+      },
+      "outputs": [],
+      "source": [
+        "dataset = tfio.IODataset.from_orc(\"iris.orc\", capacity=15).batch(1)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4xPr3f4LVdeN"
+      },
+      "source": [
+        "Examine the dataset:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "9B1QUKG70Lzs"
+      },
+      "outputs": [],
+      "source": [
+        "for item in dataset.take(1):\n",
+        "    print(item)\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "03qncHJPVNK3"
+      },
+      "source": [
+        "Let's walk through an end-to-end example of tf.keras model training with ORC dataset based on iris dataset."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tDkpKRMVcPfb"
+      },
+      "source": [
+        "### Data preprocessing"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nDgkfWFRVjKz"
+      },
+      "source": [
+        "Configure which columns are features, and which column is label:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "R1OYAybz07dr"
+      },
+      "outputs": [],
+      "source": [
+        "feature_cols = [\"sepal_length\", \"sepal_width\", \"petal_length\", \"petal_width\"]\n",
+        "label_cols = [\"species\"]\n",
+        "\n",
+        "# select feature columns\n",
+        "feature_dataset = tfio.IODataset.from_orc(\"iris.orc\", columns=feature_cols)\n",
+        "# select label columns\n",
+        "label_dataset = tfio.IODataset.from_orc(\"iris.orc\", columns=label_cols)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "GSYMP48vVvV0"
+      },
+      "source": [
+        "A util function to map species to float numbers for model training:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "TQvuE7OgVs1q"
+      },
+      "outputs": [],
+      "source": [
+        "@tf.function\n",
+        "def species_float_conversion(x):\n",
```

Comment:
This function is really strange.

1. Returns floats -> should probably return ints
2. Result is one-based
3. It's not scalable:
    * It does a linear scan to find the value instead of a dictionary lookup.
    * It can't handle batches

Either use `tf.lookup` for a tensorflow equivalent of `dict` or use `tf.keras.layers.StringLookup`/`tf.keras.layers.experimental.preprocessing`:

```
lookup = tf.keras.layers.StringLookup(
    vocabulary=["virginica","versicolor","setosa"],
     num_oov_indices=0)

lookup(["virginica","versicolor","setosa"])
```
```
<tf.Tensor: shape=(3,), dtype=int64, numpy=array([0, 1, 2])>
```

```
lookup = tf.keras.layers.StringLookup(
    vocabulary=["virginica","versicolor","setosa"],
     num_oov_indices=0, invert=True)

lookup([0,1,2,3])
```
```
<tf.Tensor: shape=(4,), dtype=string, numpy=array([b'virginica', b'versicolor', b'setosa', b'[UNK]'], dtype=object)>
```

Reply:
Thanks for shedding lights on this!